rom/laghos_rom.cpp

```diff
@@ -54,6 +54,7 @@ void ROM_Sampler::SampleSolution(const double t, const double dt, Vector const&
         if (writeSnapshots && addSample)
         {
             tSnapX.push_back(t);
+            if (pd >= 0.0) pdSnap.push_back(pd);
```

Comment:
In multiple places, there is logic like this based on whether `pd >= 0`, with negative `pd` indicating that penetration distance is not used. I suppose this is designed just for the RT problem 7, in which `pd` is always non-negative. In general, could this be negative, or is that unrealistic? Should the logic be generalized to use some flag instead of checking `pd >= 0`, or should this work for the problems we are interested in?

Reply:
I believe that's unrealistic (in physical meanings) since the initial velocity drives the flow in the anti-clockwise and the gravitation drives it downward. We have also seen in the experiments that the penetration distance at time `t` with Atwood number `A` closely fits with the theoretical finding `C*A*t^2` for some constant C. Right now the only application of an extra indicator in Laghos is the penetration distance (or `A*t^2` of the Rayleigh Taylor instability, but if we want to generalize to some other indicators in other experiments, we can do the corresponding changes easily. I lean to making these changes whenever necessary, but I can do it in this PR, or I can add comments so that our team will understand how to make the changes later.

Reply:
I agree it is fine for now.